### PR TITLE
[SMALLFIX] Minor issues w/ Object UFS

### PR DESCRIPTION
**Pull request by madanadit**
- `ObjectUnderFileSystem:getDirectoryStatus` updated to work correctly for pseudo-directory with no breadcrumb
- JOSS Swift library does not return size and last modified for pseudo-directory objects => populating `ObjectStatus` with invalid values

### CODE REVIEW

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 1]
```diff
@@ -87,16 +87,25 @@ protected ObjectUnderFileSystem(AlluxioURI uri) {
...
 
     public ObjectStatus(String name, long contentLength, long lastModifiedTimeMs) {
       mContentLength = contentLength;
       mLastModifiedTimeMs = lastModifiedTimeMs;
       mName = name;
     }
 
+    public ObjectStatus(String name) {
+      mContentLength = INVALID_CONTENT_LENGTH;
+      mLastModifiedTimeMs = INVALID_MODIFIED_TIME;
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: Do the callers know to interpret -1 as invalid?

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 2]
```diff
@@ -356,16 +365,13 @@ public long getBlockSizeByte(String path) throws IOException {
...
-    if (details != null) {
+    if (isDirectory(path)) {
       ObjectPermissions permissions = getPermissions();
       return new UfsDirectoryStatus(path, permissions.getOwner(), permissions.getGroup(),
           permissions.getMode());
-    } else {
-      LOG.error("Error fetching directory status, assuming directory does not exist");
-      throw new FileNotFoundException(path);
     }
+    LOG.error("Error fetching directory status, assuming directory does not exist");
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: Could you change the log level to `WARN` and log the path as well?

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 3]
```diff
@@ -87,16 +87,25 @@ protected ObjectUnderFileSystem(AlluxioURI uri) {
...
   protected class ObjectStatus {
-    final long mContentLength;
-    final long mLastModifiedTimeMs;
-    final String mName;
+    private final long mContentLength;
+    private final long mLastModifiedTimeMs;
+    private final String mName;
+
+    private static final long INVALID_CONTENT_LENGTH = -1L;
+    private static final long INVALID_MODIFIED_TIME = -1L;
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: static variables before instance variables?

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java (hunk 1 above)
Comment (madanadit) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: @calvinjia `contentLength` and `lastModified` are only called for files to populate `UfsFileStatus`. In that case, these are valid values.

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 4]
```diff
@@ -397,7 +403,7 @@ public UfsFileStatus getFileStatus(String path) throws IOException {
       return new UfsFileStatus(path, details.getContentLength(), details.getLastModifiedTimeMs(),
           permissions.getOwner(), permissions.getGroup(), permissions.getMode());
     } else {
-      LOG.error("Error fetching file status, assuming file does not exist");
+      LOG.warn("Error fetching file status, assuming file %s does not exist", path);
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: I think you want to use `file {} does not exist", path`?

**Pull request merged**